kong/clustering/services/sync/rpc.lua

```diff
@@ -176,7 +177,23 @@ local function is_rpc_ready()
 end
 
 
-local function do_sync()
+local do_sync
+
+
+local function error_handler(err, is_full_sync)
+  ngx_log(ngx_ERR, err)
+
+   -- retry without releasing the mutex as the DP has nothing to do except syncing
+  if is_full_sync then
+    yield()
+    return do_sync()
```

Comment:
why do we run sync in a timer? could it cause more cost?

Reply:
By doing so we do not release the lock and fetch the lock again

Reply:
Potential infinite loop?

Reply:
I understand in this state, there is nothing DP can do other than sync, but it is a little bit risky do sync again and again without any backoff delay.